### PR TITLE
feat: rework batch workflow

### PR DESCRIPTION
**Pull request by bradleyjones**
- **chore: add cover.cov to gitignore**
- **feat: rework the batching logic**

Calculate batches at the end of the inventory workflow

### CODE REVIEW

File: pkg/lib.go [hunk 1]
```diff
@@ -378,21 +367,95 @@ func GetInventoryReports(cfg *config.Application) (AccountRoutedReports, error)
...
+
+		nodeMap := make(map[string]inventory.Node)
+		for _, node := range accountReport.Nodes {
+			nodeMap[node.UID] = node
+		}
+		podMap := make(map[string]inventory.Pod)
+		for _, pod := range accountReport.Pods {
+			podMap[pod.UID] = pod
+		}
+		containersByPod := make(map[string][]inventory.Container)
```
Comment (bot190) [inline pkg/lib.go:415]: Is this happening after the data has been sanitized? Don't want to have an error here if we're missing data.

File: pkg/lib.go [hunk 2]
```diff
@@ -378,21 +367,95 @@ func GetInventoryReports(cfg *config.Application) (AccountRoutedReports, error)
...
+				batchedContainersSlice = append(batchedContainersSlice, containersByNamespace[ns.UID]...)
+				for _, pod := range batchedPodsSlice {
+					batchedNodesMap[pod.NodeUID] = nodeMap[pod.NodeUID]
 				}
-				reports[account] = append(reports[account], accountReport)
 			}
+			batchedNodesSlice := []inventory.Node{}
+			for _, node := range batchedNodesMap {
+				batchedNodesSlice = append(batchedNodesSlice, node)
+			}
```
Comment (bot190) [inline pkg/lib.go:444]: Is this just converting the map to an array? There isn't a cleaner way to do this?

File: pkg/lib_test.go [hunk 3]
```diff
@@ -292,3 +318,463 @@ func TestGetNamespacesBatches(t *testing.T) {
...
+						Namespaces:  []inventory.Namespace{TestNamespace5},
+						Containers:  []inventory.Container{Container5},
+						Pods:        []inventory.Pod{Pod5},
+						Nodes:       []inventory.Node{Node3},
+					},
+				},
+			},
+		},
+		{
+			name: "multiple batches (individual namespaces)",
```
Comment (bot190) [inline pkg/lib_test.go:622]: This looks the same as the previous test case?

File: pkg/lib.go (hunk 1 above)
Comment (bradleyjones) [reply pkg/lib.go:415]: no the sanitization of the report happens in (well just before) the post so shouldn't be any cause for concern

File: pkg/lib.go (hunk 2 above)
Comment (bradleyjones) [reply pkg/lib.go:444]: not really, there is a Values function I could call but it does this under the hood. https://cs.opensource.google/go/x/exp/+/fc45aab8:maps/maps.go;l=20

File: pkg/lib_test.go (hunk 3 above)
Comment (bradleyjones) [reply pkg/lib_test.go:622]: I does indeed appear to be the same, must have fat fingered it I'll delete it

**Review by bot190 — APPROVED** [review]

File: pkg/lib.go (hunk 2 above)
Comment (bot190) [reply pkg/lib.go:444]: That might be a bit cleaner but no perf gain so this is fine.

**Pull request merged**